chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -695,4 +695,35 @@ std::vector<Array> Split(const Array& ary, std::vector<int64_t> indices, int8_t
     return out;
 }
 
+Array Swapaxes(const Array& ary, int8_t axis1, int8_t axis2) {
+    Shape shape = ary.shape();
+    Strides strides = ary.strides();
+
+    // for negative axis.
+    axis1 = axis1 >= 0 ? axis1 : ary.ndim() + axis1;
+    axis2 = axis2 >= 0 ? axis2 : ary.ndim() + axis2;
```

Comment:
We have a function that does exactly this and the error checking https://github.com/chainer/chainer/blob/master/chainerx_cc/chainerx/axes.h#L85.